components/CreateOrUpdateRecipeForm.js

```diff
@@ -0,0 +1,212 @@
+import styled from "styled-components";
+import { useRouter } from "next/router";
+import Link from "next/link";
+
+export default function CreateOrUpdateRecipeForm({
+  errorMessage,
+  inputValidation,
+  recipeDetails,
```

Comment:
```suggestion
  recipeDetails = {},
```
You could consider giving the recipeDetails an empty object as default value like this so you wouldn't have to conditionally check whether an input has a value (when updating) or not (when creating).  `recipeDetails.name` would give you an empty input field in your create form because the form doesn't get a recipeDetails prop and sets the value as empty. 
At least this is how Diego did it in the CRUD workshop: [Services Project](https://github.com/neuefische/web-exercises/blob/crud-recap/examples/crud-recap/components/Forms/ServiceForm/index.js) 😄